app/controllers/submissions_controller.rb

```diff
@@ -277,6 +277,12 @@ def browse
     @groupings = Grouping.get_groupings_for_assignment(@assignment,
                                                        current_user)
     @sections = Section.order(:name)
+
+    @tas = Set.new
+    @groupings.map do |grouping|
+      grouping.tas.pluck(:user_name).each{|ta| @tas.add(ta)}
```

Comment:
I think there's a more elegant way of doing this - `@assignment.ta_memberships` should work, and from there you can join with `:tas` and then pluck `ta.user_name`